src/interceptors/ClientRequest/utils/normalizeHttpRequestParams.ts

```diff
@@ -67,6 +72,30 @@ export function normalizeHttpRequestParams(
 
     callback = resolveCallback(args)
   }
+  // Handle a legacy Url and re-normalize from either a RequestOptions object
+  // or a WHATWG URL
+  else if ('hash' in args[0] && !('method' in args[0])) {
+    if (args[0].hostname === null) {
+      /*
+        We are dealing with a relative url, so use the path as an "option" and
+        merge-in any existing options, giving priority to exising options -- i.e. a path in any
+        existing options will take precedence over the one contained in the url. This is consistent
+        with the behaviour in ClientRequest.
+        
+        https://github.com/nodejs/node/blob/d84f1312915fe45fe0febe888db692c74894c382/lib/_http_client.js#L122
+      */
+      debug('given a relative legacy url:', args[0])
+
+      return isRequestOptions(args[1])
+        ? normalizeHttpRequestParams({path: args[0].path, ...args[1]}, args[2])
+        : normalizeHttpRequestParams({path: args[0].path}, args[1] as HttpRequestCallback)
+    }
+
+    debug('given an absolute legacy url:', args[0])
+
+    //We are dealing with an absolute url, so convert to WHATWG and try again
+    return normalizeHttpRequestParams(...[new URL(args[0].href), ...args.slice(1)] as HttpRequestArgs)
```

Comment:
This argument is quite hard to read through. Do you think we could break it down into pieces?

Reply:
I've added some checks here to allow the types to be inferred by typescript, instead.